text/0000-serialize-deserialize.md

```diff
@@ -0,0 +1,174 @@
+* Start Date: 2018-08-01
+* RFC PR: https://github.com/fusionjs/rfcs/pull/8
+
+# Summary
+
+This RFC describes a process for standardizing on an API for serializing and deserializing state between server and browser plugins.
+
+# Basic example
+
+```js
+import {SharedStateToken, createPlugin} from 'fusion-core';
+
+export default createPlugin({
+  deps: {
+    SharedState: SharedStateToken,
+  },
+  middleware: ({SharedState}) => {
+    const {serialize, deserialize} = SharedState('some-id');
+    return (ctx, next) => {
+      if (__BROWSER__) {
+        // load and parse serialized data from dom node
+        const data = deserialize();
+      } else {
+        // serialize json data into body
+        serialize(ctx, {
+          some: 'json',
+        });
+      }
+      return next();
+    };
+  },
+});
+
+// in tests
+import {SharedStateToken, MockSharedState} from 'fusion-core';
+app.register(SharedStateToken, MockSharedState({
+  'some-id': {
+    some: 'json'
+  },
+  'other-data': {}
+}));
+```
+
+# Motivation
+
+It is a very common pattern in fusion to have some shared state between the server and
```

Comment:
What types of shared data are we expecting to support? Should you expect to serialize your entire redux state tree? Should you use it for storing preferences? Or how agnostic should it be to your data type?

Reply:
By default this will just support JSON serializable data, but since the serializer/deserializer is in the DI system, you can basically add support for any data type you want.